### ISSUE TITLE
Make CupertinoPicker accessible

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#20561]
Edit:
Currently cupertino picker directly exposes the scrollable list of children. Instead it needs to describe it's semantics as an adjustable, with values set appropriately and all child semantics hidden.

_____
Because the CupertinoPicker can take an arbitrary list of widgets, determining the current, previous, and next accessibility labels for the adjustable semantics is non-trivial. Consider the following snippet of code to make this work currently:

```dart
class RenderCupertinoSemantics extends RenderProxyBox {
 ...
@override
  void assembleSemanticsNode(SemanticsNode node, SemanticsConfiguration config, Iterable<SemanticsNode> children) {
    if (children.isEmpty)
      return super.assembleSemanticsNode(node, config, children);
    final SemanticsNode scrollable = children.first;
    final List<SemanticsNode> labels = <SemanticsNode>[];
    scrollable.visitChildren((SemanticsNode child) {
      labels.add(child);
      return true;
    });
    int middle = 0;
    for (int i = 0; i < labels.length; i++) {
      final SemanticsNode label = labels[i];
      final double offset = label.transform.getTranslation()[1];
      if (offset > _kMagicMinOffset && offset < _kMagicMaxOffset) {
        middle = i;
      }
    }
    config.value = labels[middle].label;
    if (middle + 1 < labels.length)
      config.increasedValue = labels[middle + 1].label;
    if (middle - 1 >= 0)
      config.decreasedValue = labels[middle - 1].label;
    super.assembleSemanticsNode(node, config, const <SemanticsNode>[]);
  }
}

```

The only way to find the values is to traverse the set of built SemanticsNodes and use the scroll offset to guestimate which one came from the center. 

A data type like `CupertinoPickerItem` or `CupertinoPickerItem<T>` could also encapsulate the offsetting rules, while providing a getter which returned a string representation we could use for semantics. Then the above calculation would just be:

```dart
  @override
  void describeSemanticsConfiguration(SemanticsConfiguration config) {
    super.describeSemanticsConfiguration(config);
    config.isSemanticBoundary = true;
    config.value = child[_selectedItem].value.toString();
    if (_childCount != null && _selectedItem < _childCount - 1) {
      config.onIncrease = _handleIncrease;
      config.value = child[_selectedItem + 1].value.toString();
    }
    if (_selectedItem > 0) {
      config.onDecrease = _handleDecrease;
      config.value = child[_selectedItem-+ 1].value.toString();
    }
    config.textDirection = textDirection;
  }
```

cc @matthew-carroll , @xster

**Comment by jonahwilliams** [comment]
@matthew-carroll no, The Cupertino pickers have very specific a11y behavior which require us to always know the current, prev, and next items for accessibility. An arbitrary collection widget, which may also be lazy, doesn't have this requirement and the existing semantics merging handles this.

**Comment by jonahwilliams** [comment]
Another approach might requiring a callback function which, given an index, provides the accessibility label for that index, or null.  The issue with this is that it becomes easier to make a mistake, either by providing the wrong value or skipping it altogether (it's really a requirement for this widget).

**Comment by jonahwilliams** [comment]
I've prepared an example of the breaking changes here: https://github.com/flutter/flutter/pull/20599

What I still don't know is what would be the ideal mapping from CupertinoPickerItem to an actual widget.

**Comment by matthew-carroll** [comment]
What's the problem with the existing code that you're trying to solve?

**Comment by jonahwilliams** [comment]
It's not accessible :)

**Comment by jonahwilliams** [comment]
@xster and I spoke offline, I'm going to try a different approach that is non-breaking. Repurposing this issue for the overal a11y issue.
